### PR TITLE
Better .npmignore + Allow ES6 sources

### PR DESCRIPTION
**Pull request by nkbt**
Tested with local install
```
[nkbt@nkbt] tmp $ npm install ~/github/react-immutable-proptypes
 
> react-immutable-proptypes@1.2.1 prepublish /Users/nkbt/github/react-immutable-proptypes
> npm run build

/
> react-immutable-proptypes@1.2.1 build /Users/nkbt/github/react-immutable-proptypes
> ./scripts/build

src/ImmutablePropTypes.js -> dist/ImmutablePropTypes.js
src/__tests__/ImmutablePropTypes-test.js -> dist/__tests__/ImmutablePropTypes-test.js
react-immutable-proptypes@1.2.1 node_modules/react-immutable-proptypes


[nkbt@nkbt] tmp $ find node_modules/react-immutable-proptypes
node_modules/react-immutable-proptypes
node_modules/react-immutable-proptypes/CHANGELOG.md
node_modules/react-immutable-proptypes/dist
node_modules/react-immutable-proptypes/dist/ImmutablePropTypes.js
node_modules/react-immutable-proptypes/LICENSE
node_modules/react-immutable-proptypes/package.json
node_modules/react-immutable-proptypes/README.md
node_modules/react-immutable-proptypes/src
node_modules/react-immutable-proptypes/src/ImmutablePropTypes.js
```

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -1 +1,4 @@
-src
```
Comment (nkbt) [inline .npmignore]: Most of people now compile their scripts (me included) so it is good to let them use ES6 directly

**Comment by HurricaneJames** [comment]
I am uncertain how this PR helps the package. Does it just include the source in the npmjs.com distributed package? I cannot see how this would hurt anything, but I am uncertain of the advantages in this case. Can you provide further details how how this would help, please? If for no other reason than I might learn something new that would make my own development practices better. Thanks!

**Comment by nkbt** [comment]
1. Add ES6 source to npm package, so it is possible to use it and compile manually:
  ```
  import {contains} from 'react-immutable-proptypes/src'
  ```
  Then I have full control over sourcemaps and so on. Easier to debug and develop.

2. Remove all unneeded files from npm package. It is not a very good practice to ship your dev stuff to the package. It is simply unnecessary (and adds extra bytes too).

**Comment by nkbt** [comment]
Addition to `2`
Now installation of the package looks like this:

```
✔ ~/www/frontend [bugfix L|✚ 5…1] 
00:43 $ find node_modules/react-immutable-proptypes
node_modules/react-immutable-proptypes
node_modules/react-immutable-proptypes/.eslintrc
node_modules/react-immutable-proptypes/.idea
node_modules/react-immutable-proptypes/.idea/.name
node_modules/react-immutable-proptypes/.idea/encodings.xml
node_modules/react-immutable-proptypes/.idea/jsLibraryMappings.xml
node_modules/react-immutable-proptypes/.idea/misc.xml
node_modules/react-immutable-proptypes/.idea/modules.xml
node_modules/react-immutable-proptypes/.idea/react-immutable-proptypes.iml
node_modules/react-immutable-proptypes/.idea/vcs.xml
node_modules/react-immutable-proptypes/.idea/workspace.xml
node_modules/react-immutable-proptypes/.istanbul.yml
node_modules/react-immutable-proptypes/.jshintrc
node_modules/react-immutable-proptypes/.npmignore
node_modules/react-immutable-proptypes/.nvmrc
node_modules/react-immutable-proptypes/.travis.yml
node_modules/react-immutable-proptypes/babel.js
node_modules/react-immutable-proptypes/CHANGELOG.md
node_modules/react-immutable-proptypes/dist
node_modules/react-immutable-proptypes/dist/__tests__
node_modules/react-immutable-proptypes/dist/__tests__/ImmutablePropTypes-test.js
node_modules/react-immutable-proptypes/dist/ImmutablePropTypes.js
node_modules/react-immutable-proptypes/LICENSE
node_modules/react-immutable-proptypes/package.json
node_modules/react-immutable-proptypes/README.md
node_modules/react-immutable-proptypes/scripts
node_modules/react-immutable-proptypes/scripts/build
node_modules/react-immutable-proptypes/scripts/test
node_modules/react-immutable-proptypes/scripts/test-cov

```

**Comment by nkbt** [comment]
@HurricaneJames what is the status of PR?

**Comment by HurricaneJames** [comment]
Sorry for the delay, I was really busy with other stuff the last few days. It is merged and published as v1.2.3. Thanks.

**Comment by nkbt** [comment]
No problems! Thanks you!

**Pull request merged**